pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotRealtimeTableResource.java

```diff
@@ -136,15 +136,22 @@ public Response resumeConsumption(
       notes = "Force commit the current segments in consuming state and restart consumption. "
           + "This should be used after schema/table config changes. "
           + "Please note that this is an asynchronous operation, "
-          + "and 200 response does not mean it has actually been done already")
+          + "and 200 response does not mean it has actually been done already."
+          + "If specific partitions or consuming segments are provided, "
+          + "only those partitions or consuming segments will be force committed.")
   public Map<String, String> forceCommit(
-      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName) {
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
+      @ApiParam(value = "Comma separated list of partition group IDs to be committed") @QueryParam("partitions")
+      String partitionGroupIds,
+      @ApiParam(value = "Comma separated list of consuming segments to be committed") @QueryParam("segments")
```

Comment:
let us modify this documentation as well

Reply:
I changed the variableName in the method as well as the documentation, but for the actual API's query param, I believe the shorter forms - `partitions` and `segments` - are better as they make the endpoint shorter:
```
POST /tables/{tableName}/forceCommit?segments=s1,s2
```
instead of
```
POST /tables/{tableName}/forceCommit?segmentsToCommit=s1,s2
```

Reply:
Wait. Why do we have both partitiongroupIds as well as consuming segments? There is only one consuming segment per partition, so it is just enough to provide the segment names right?

Reply:
Having the option to provide either makes it easier for operators.
Keep in mind that both params cannot be provided at the same time, we fail the request. It's either partitions or segments.